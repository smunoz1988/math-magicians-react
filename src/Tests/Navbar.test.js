import React from 'react';
import {
  render, screen, act, fireEvent,
} from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import NavBar from '../Components/Navbar';

describe('NavBar', () => {
  it('renders correctly', () => {
    const { container } = render(
      <BrowserRouter>
        <NavBar />
      </BrowserRouter>,
    );

    expect(screen.getByText('Math Magicians')).toBeInTheDocument();
    expect(screen.getByText('Home')).toBeInTheDocument();
    expect(screen.getByText('Calculator')).toBeInTheDocument();
    expect(screen.getByText('Quotes')).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });

  it('navigates to the Calculator page when Calculator link is clicked', () => {
    const { container } = render(
      <BrowserRouter>
        <NavBar />
      </BrowserRouter>,
    );

    act(() => {
      const calculatorLink = screen.getByText('Calculator');
      fireEvent.click(calculatorLink);
    });

    expect(window.location.pathname).toBe('/Calculator');

    expect(container.firstChild).toMatchSnapshot();
  });
});
