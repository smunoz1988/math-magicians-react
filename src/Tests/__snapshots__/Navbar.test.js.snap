// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar navigates to the Calculator page when Calculator link is clicked 1`] = `
<nav>
  <h2>
    Math Magicians
  </h2>
  <div
    class="navItemsContainer"
  >
    <a
      class="navItems"
      href="/"
    >
      Home
    </a>
    <a
      aria-current="page"
      class="navItems active"
      href="/Calculator"
    >
      Calculator
    </a>
    <a
      class="navItems"
      href="/Quotes"
    >
      Quotes
    </a>
  </div>
</nav>
`;

exports[`NavBar renders correctly 1`] = `
<nav>
  <h2>
    Math Magicians
  </h2>
  <div
    class="navItemsContainer"
  >
    <a
      class="navItems"
      href="/"
    >
      Home
    </a>
    <a
      class="navItems"
      href="/Calculator"
    >
      Calculator
    </a>
    <a
      class="navItems"
      href="/Quotes"
    >
      Quotes
    </a>
  </div>
</nav>
`;
